lib/premailer/adapter/rgb_to_hex.rb

```diff
@@ -9,13 +9,14 @@ def to_hex(str)
     def is_rgb?(color)
       pattern = %r{
       rgb
-      \(\s*    # literal open, with optional whitespace
-      (\d{1,3})   # capture 1-3 digits
-      \s*,\s*     # comma, with optional whitespace
-      (\d{1,3})   # capture 1-3 digits
-      \s*,\s*     # comma, with optional whitespace
-      (\d{1,3})   # capture 1-3 digits
-      \s*\)       # literal close, with optional whitespace
+      \(\s*                  # literal open, with optional whitespace
+      (\d{1,3})              # capture 1-3 digits
+      \s*,?\s*               # optional comma, with optional whitespace
```

Comment:
should it be something like that because `12` also matches `\d{1,3}\s*,?\d{1,3}`
```suggestion
      (?:\s*,\s*|\s+)               # optional comma, with optional whitespace
```

```
>> "12" =~ /\d{1,3}\s*,?\d{1,3}/
=> 0
```

Reply:
Good catch. Updated. Also added percentage opacity modifiers.

Note that this doesn't implement all of the [CSS Colors 4 definition for `rgb`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb), namely percentage values (`rgb(30% 20% 50%);`). It also lops off the alpha value from the RGB instead of specifying the Alpha by continuing to build a 6 digit (as opposed to [8 digit hex code](https://caniuse.com/css-rrggbbaa). Selfishly since this is premailer and email targets old rendering engines, I'd rather have it drop the alpha value anyway (since neither alpha nor 8 digit hex codes are typically supported)